src/main/scala/tech/cryptonomic/conseil/tezos/TezosDatabaseOperations.scala

```diff
@@ -156,64 +143,94 @@ object TezosDatabaseOperations {
     }
 
   /**
-    * Generates database rows from a list of averageFees calculated for some operation kinds.
-    * @param maybeFees  List of possible new fees that may have been calculated for different
-    *                   operation kinds.
-    * @return           Database rows
+    *
+    * @param fees      List of average fees for different operation kinds.
+    * @return          Database action possibly containing the number of rows written (if available from the underlying driver)
     */
-  def feesToDatabaseRows(maybeFees: List[Option[AverageFees]]): List[Tables.FeesRow] = {
-    maybeFees
-      .filter(_.isDefined)
-      .map{ someFee =>
-        val fee = someFee.get
-        Tables.FeesRow(
-          low = fee.low,
-          medium = fee.medium,
-          high = fee.high,
-          timestamp = fee.timestamp,
-          kind = fee.kind
-        )
-      }
-  }
-
+  def writeFeesIO(fees: List[AverageFees]): DBIO[Option[Int]] =
+    Tables.Fees ++= fees.map(RowConversion.ofAverageFees)
+  
   /**
     * Given the operation kind, return range of fees and timestamp for that operation.
     * @param kind  Operation kind
     * @return      The average fees for a given operation kind, if it exists
     */
-  def calculateAverageFees(kind: String): Option[AverageFees] = {
-    val operationKinds = Set[String]{kind}
-    val action = for {
-      o <- Tables.Operations
-      if o.kind.inSet(operationKinds)
-    } yield (o.fee, o.timestamp)
-    val op = dbHandle.run(action.distinct.sortBy(_._2.desc).take(numberOfFeesAveraged).result)
-    val results = Await.result(op, Duration.apply(awaitTimeInSeconds, SECONDS))
-    val resultNumbers = results.map(x => x._1.getOrElse("0").toDouble)
-    val m: Int = ceil(mean(resultNumbers)).toInt
-    val s: Int = ceil(stdev(resultNumbers)).toInt
-    results.length match {
-      case 0 => None
-      case _ =>
-        val timestamp = results.head._2
-        Some(AverageFees(max(m - s, 0), m, m + s, timestamp, kind))
+  def calculateAverageFeesIO(kind: String)(implicit ec: ExecutionContext): DBIO[Option[AverageFees]] = {
+    def computeAverage(ts: java.sql.Timestamp, fees: Seq[(Option[String], java.sql.Timestamp)]): AverageFees = {
+      val values = fees.map {
+        case (fee, _) => fee.map(_.toDouble).getOrElse(0.0)
+      }
+      val m: Int = ceil(mean(values)).toInt
+      val s: Int = ceil(stdev(values)).toInt
+      AverageFees(max(m - s, 0), m, m + s, ts, kind)
     }
-  }
 
+    val opQuery =
+      Tables.Operations
+        .filter(_.kind === kind)
```

Comment:
Oh wow, this might be a way simpler filter than what was originally there. Did you test this with postman to see if both versions gave the same result?

Reply:
will do soon

Reply:
I tried to verify the fees calculation on the staging zeronet environment, but it proved impractical. There are more than 85000 block levels and getting data from a remote connection takes much more than a local node, which would mean that to download and process the Conseil db from scratch seemed to be taking hours.

I did a different test, by using a locally started alphanet node, and running Lorre over a scratch database instance to load all data and make the fee computation, once on this PR's branch and once on the `master`.
This proved more reasonable and the results loaded on postman calling a local Conseil instance are exactly the same.

Reply:
These are the results for non-zero operation types, so you can check if they seems to be consistent with real results

```json
{
    "low": 0,
    "medium": 17110,
    "high": 111665,
    "timestamp": 1534163782000,
    "kind": "transaction"
}

{
    "low": 32435,
    "medium": 46000,
    "high": 59565,
    "timestamp": 1534121172000,
    "kind": "origination"
}

{
    "low": 14095,
    "medium": 36364,
    "high": 58633,
    "timestamp": 1534162102000,
    "kind": "delegation"
}
```

Reply:
results using data from `staging/zeronet`, limited to 1000 blocks, comparing once more
branch `CONSEIL-88-async-db-calls` vs `master`

results are identical once more
```
low  |medium |high  |timestamp           |kind                  |
-----|-------|------|--------------------|----------------------|
0    |0      |0     |2018-07-21 10:03:27 |seed_nonce_revelation |
0    |25000  |50000 |2018-07-21 10:02:47 |delegation            |
9763 |33334  |56905 |2018-07-21 10:01:27 |transaction           |
0    |0      |0     |2018-07-21 10:00:07 |activate_account      |
0    |0      |0     |2018-07-21 10:51:47 |origination           |
0    |0      |0     |2018-07-21 10:27:07 |reveal                |
0    |0      |0     |2018-07-21 10:51:47 |endorsement           |
```

Reply:
@ivanopagano You can also point your conf file to the staging database if you are simply testing Conseil changes.

Reply:
I'd need the staging db conf in that case, but here we're writing fees to it so this case is not safe to test against staging